checkstyle-tester/README.md

```diff
@@ -1,138 +1,61 @@
-=============================================
-
-## [diff.groovy] Diff report generation
-
-In order to generate a compact diff report before and after your changes you can use diff.groovy script which performs all required work automatically. Note, diff.groovy ignores excludes specified in the list of projects file.
-Please execute the following command in your command line to run diff.groovy:
+If you want to specify different Checkstyle configs for base branch and patch branch use the following command:
 
 ```
-groovy diff.groovy --localGitRepo /home/johndoe/projects/checkstyle --baseBranch master --patchBranch i111-my-fix --config my_check.xml --listOfProjects projects-to-test-on.properties
+groovy diff.groovy --localGitRepo /home/johndoe/projects/checkstyle --baseBranch master --patchBranch i111-my-fix --baseConfig base_config.xml --patchConfig patch_config.xml --listOfProjects projects-to-test-on.properties
 ```
 
 or with short command line arguments names:
 
 ```
-groovy diff.groovy -r /home/johndoe/projects/checkstyle -b master -p i111-my-fix -c my_check.xml -l projects-to-test-on.properties
+groovy diff.groovy -r /home/johndoe/projects/checkstyle -b master -p i111-my-fix -bc base_config.xml -pc patch_config.xml -l projects-to-test-on.properties
 ```
 
-If you want to specify different Checkstyle configs for base branch and patch branch use the following command:
+To generate the report only for the patch branch which contains your changes, use the following command:
 
 ```
-groovy diff.groovy --localGitRepo /home/johndoe/projects/checkstyle --baseBranch master --patchBranch i111-my-fix --baseConfig base_config.xml --patchConfig patch_config.xml --listOfProjects projects-to-test-on.properties
+groovy diff.groovy --localGitRepo /home/johndoe/projects/checkstyle --patchBranch i111-my-fix --patchConfig patch_config.xml --listOfProjects projects-to-test-on.properties --mode single
 ```
 
 or with short command line arguments names:
 
 ```
-groovy diff.groovy -r /home/johndoe/projects/checkstyle -b master -p i111-my-fix -bc base_config.xml -pc patch_config.xml -l projects-to-test-on.properties
+groovy diff.groovy -r /home/johndoe/projects/checkstyle -p i111-my-fix -pc patch_config.xml -l projects-to-test-on.properties -m single
 ```
 
 The script receives the following set of command line arguments:
 
 **localGitRepo** (r) - path to the local Checkstyle repository (required);
 
-**baseBranch** (b) - name of the base branch in local Checkstyle repository (optional, default is master branch);
+**baseBranch** (b) - name of the base branch in local Checkstyle repository (optional, if asent, then the tool will use only patchBranch in case the tool mode is 'single', otherwise baseBrach will be set to 'master');
+
+**mode** (m) - The mode of the tool: 'single' or 'diff'. Set this option to 'single' if your patch branch contains changes for any option that can't work on master/base branch. For example, for new properties, new tokens, or new modules.
```

Comment:
> I don't see a mention if this is required or optional.
We should mention that diff is the default?

It was mentioned at the end of the property description. I moved it at the start of the option description.

> After the sentence "For example, ...", please add a sentence that says something like "For all other changes, 'diff' mode should be the preferred mode used." so it is clear.

Done.